ota-plus-web/app/reactapp/src/components/updates/UpdateDetails.jsx

```diff
@@ -30,91 +38,96 @@ class UpdateDetails extends Component {
                             placeholder="Type here"
                             rows={ 5 }
                             name="updateDescription"
-                            id={ "update-description" + details.name }
-                            defaultValue={ details.description }
+                            id={ "update-description" + updateItem.name }
+                            defaultValue={ updateItem.description }
                             isEditable={ isEditable }
                         />
                     </div>
                 </div>
+                <div className="row targets-container">
                 {
-                    !details.targets ?
-                        details.source &&
-                        <div className="row source-container">
-                            <div className="col-xs-12">
-                                <label className="c-form__label">{ details.source.id }</label>
-                                { "No source information available." }
-                            </div>
+                    updatesStore.updatesFetchMtuIdAsync.isFetching ?
+                        <div className="wrapper-center">
+                            <Loader/>
                         </div>
                         :
-                        <div className="row targets-container">
-                        {
-                            _.map(details.targets, (target, hardwareId) => {
+                        mtuData ?
+                            _.map(mtuData, (target, hardwareId) => {
+                                const noInformation = "No information.";
+                                const fromPackage = target.from.target;
```

Comment:
@meedhoas I suggest to remove these variables and make shorter declaration using destructing. It's up to you.
`const {target:fromPackage, checksum:fromVersion} = target.from;
 const {target:toPackage, checksum:toVersion} = target.to;`